airflow/gcp/operators/search_ads.py

```diff
@@ -0,0 +1,200 @@
+        :ref:`howto/operator:GoogleSearchAdsGetfileReportOperator`
+
+    :param report_id: ID of the report.
+    :type report_id: str
+    :param bucket_name: The bucket to upload to.
+    :type bucket_name: str
+    :param report_name: The report name to set when uploading the local file. If not provided then
+        report_id is used.
+    :type report_name: str
+    :param gzip: Option to compress local file or file data for upload
+    :type gzip: bool
+    :param api_version: The version of the api that will be requested for example 'v3'.
+    :type api_version: str
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate, if any. For this to work, the service accountmaking the
+        request must have  domain-wide delegation enabled.
+    :type delegate_to: str
+    """
+
+    template_fields = ("report_name", "report_id", "bucket_name")
+
+    @apply_defaults
+    def __init__(
+        self,
+        report_id: str,
+        bucket_name: str,
+        report_name: Optional[str] = None,
+        gzip: bool = True,
+        chunk_size: int = 10 * 1024 * 1024,
+        api_version: str = "v2",
+        gcp_conn_id: str = "google_cloud_default",
+        delegate_to: Optional[str] = None,
+        *args,
+        **kwargs
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.report_id = report_id
+        self.api_version = api_version
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+        self.report_id = report_id
+        self.chunk_size = chunk_size
+        self.gzip = gzip
+        self.bucket_name = self._set_bucket_name(bucket_name)
+        name = report_name or report_id
+        self.report_name = self._set_report_name(name)
+
+    @staticmethod
+    def _set_report_name(name: str) -> str:
+        return name if name.endswith(".csv") else name + ".csv"
+
+    @staticmethod
+    def _set_bucket_name(name: str) -> str:
+        bucket = name if not name.startswith("gs://") else name[5:]
+        return bucket.strip("/")
+
+    @staticmethod
+    def _handle_report_fragment(fragment: bytes) -> bytes:
+        fragment_records = fragment.split(b"\n", 1)
+        if len(fragment_records) > 1:
+            return fragment_records[1]
+        return b""
+
+    def execute(self, context: Dict):
+        hook = GoogleSearchAdsHook(
+            gcp_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to,
+            api_version=self.api_version,
+        )
+
+        gcs_hook = GoogleCloudStorageHook(
+            gcp_conn_id=self.gcp_conn_id, delegate_to=self.delegate_to
+        )
+
+        self.log.info("Downloading Search Ads report %s", self.report_id)
+
+        # Resolve report fragments
+        response = hook.get(report_id=self.report_id)
+        fragments = range(len(response["files"]))
```

Comment:
```suggestion
        fragments_count = range(len(response["files"]))
```
This variable name is confusing.